### PR TITLE
Add grpc-gateway support

### PR DESCRIPTION
**Pull request by heartless-clown**
Adds support for protoc-gen-grpc-gateway and protoc-gen-grpc-openapiv2 plugins.

### CODE REVIEW

File: example/BUILD.bazel [hunk 1]
```diff
@@ -20,13 +20,40 @@
...
-# gazelle:proto_rule proto_go_library deps @org_golang_google_protobuf//reflect/protoreflect
-# gazelle:proto_rule proto_go_library deps @org_golang_google_protobuf//runtime/protoimpl
+# gazelle:proto_plugin protoc-gen-go dep @org_golang_google_protobuf//reflect/protoreflect
+# gazelle:proto_plugin protoc-gen-go dep @org_golang_google_protobuf//runtime/protoimpl
 # gazelle:proto_rule proto_go_library visibility //visibility:public
 # gazelle:proto_language go plugin protoc-gen-go
 # gazelle:proto_language go rule proto_compile
-## Hold off on proto_go_library until we upgrade the deps of this repo
-## gazelle:proto_language go rule proto_go_library
+# TODO: Investigate why build example/routeguide:routeguide_go_proto fails:
```
Comment (pcj) [inline example/BUILD.bazel:28]: Is this still an issue?

File: example/BUILD.bazel [hunk 2]
```diff
@@ -20,13 +20,40 @@
 ## Go ##
 # gazelle:proto_plugin protoc-gen-go implementation golang:protobuf:protoc-gen-go
 # gazelle:proto_rule proto_go_library implementation stackb:rules_proto:proto_go_library
-# gazelle:proto_rule proto_go_library deps @org_golang_google_protobuf//reflect/protoreflect
-# gazelle:proto_rule proto_go_library deps @org_golang_google_protobuf//runtime/protoimpl
+# gazelle:proto_plugin protoc-gen-go dep @org_golang_google_protobuf//reflect/protoreflect
```
Comment (pcj) [inline example/BUILD.bazel:23]: What was the reasoning around this change?

File: example/routeguide/BUILD.bazel [hunk 3]
```diff
@@ -129,18 +129,54 @@ proto_compile(
...
+        "@build_stack_rules_proto//plugin/grpc-ecosystem/grpc-gateway:protoc-gen-grpc-openapiv2",
+        "@build_stack_rules_proto//plugin/grpc/grpc-go:protoc-gen-go-grpc",
+    ],
     proto = "routeguide_proto",
 )
 
 proto_go_library(
     name = "routeguide_go_proto",
-    srcs = ["routeguide.pb.go"],
+    srcs = [
```
Comment (pcj) [inline example/routeguide/BUILD.bazel:163]: A separate example is probably warranted for grpc-gateway rather than adding it to routeguide

**Review by pcj — CHANGES_REQUESTED** [review]
I think a separate example/golden/testdata/grpc-gateway would be helpful here.  Also, needs rebase once https://github.com/stackb/rules_proto/pull/252 is landed.

File: example/routeguide/BUILD.bazel (hunk 3 above)
Comment (heartless-clown) [reply example/routeguide/BUILD.bazel:163]: I'll try to provide a separate /golden/testdata/grpc-gateway example as well as a working (well, maybe not really, as the routeguide_go_proto build doesn't work) golang example.

**Comment by pcj** [comment]
No longer needed?

**Pull request closed without merging**